add shader quad control sample demonstrating VK_KHR_shader_quad_control features.

gpx1000 opened the pull request:
## Description

This adds in the shader quad control sample.

## General Checklist:

Please ensure the following points are checked:

- [ ] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [ ] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [ ] I have commented any added functions (in line with Doxygen)
- [ ] I have commented any code that could be hard to understand
- [ ] My changes do not add any new compiler warnings
- [ ] My changes do not add any new validation layer errors or warnings
- [ ] I have used existing framework/helper functions where possible
- [ ] My changes do not add any regressions
- [ ] I have tested every sample to ensure everything runs correctly
- [ ] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [ ] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

 If this PR contains framework changes:
 - [ ] I did a full batch run using the `batch` command line argument to make sure all samples still work properly

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the readme of the folder that the sample belongs to [e.g. api samples readme](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/samples/api/README.adoc)
- [ ] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/samples/extensions/conditional_rendering)
- [ ] For new samples, I have added a link to the [Antora navigation](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/antora/modules/ROOT/nav.adoc) so that the sample will be listed at the Vulkan documentation site

gpx1000 committed:
Reapply "add shader quad control sample demonstrating VK_KHR_shader_quad_control features."

This reverts commit 6021c012a1cc6844d0d87b0025556ad7e6432f89.

SaschaWillems commented on samples/extensions/shader_quad_control/README.adoc:
```diff
@@ -0,0 +1,80 @@
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+////
```
This is missing the "Where to get the source" Antora block (see readme.adoc for other samples or the readme from the samples template).

gpx1000 committed:
update readme for antora code link.

asuessenbach commented on samples/extensions/shader_quad_control/shader_quad_control.cpp:
```diff
@@ -0,0 +1,190 @@
+
+	VkPipelineDepthStencilStateCreateInfo depth_stencil{VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO};
+	depth_stencil.depthTestEnable  = VK_FALSE;
+	depth_stencil.depthWriteEnable = VK_FALSE;
```
Why do you emphasize that no depth test or writing is done here?
You actually don't **need** to explicitly set those members to false.

gpx1000 replied:
Force of habit I guess :)

asuessenbach reviewed (commented):
Just one minor issue on source level.

Could you add a picture how it's supposed to look like?
I'm getting a window with seemingly perfectly smooth color gradient.

gary-sweet reviewed (commented):
All I can say is that this correctly reports as unsupported for me.

gpx1000 committed:
Merge branch 'main' into shader_quad_control

# Conflicts:
#	samples/extensions/README.adoc

gpx1000 committed:
Remove unnecessary depth stencil state configuration

The depth test and write enable flags are already false by default after zero-initialization.

gpx1000 committed:
Merge branch 'main' into shader_quad_control

gpx1000 committed:
Merge branch 'main' into shader_quad_control

# Conflicts:
#	framework/vulkan_type_mapping.h
#	samples/extensions/README.adoc

gpx1000 committed:
fix for merge conflicts.

gpx1000 committed:
Update copyright year to 2025-2026 and align formatting in shader_quad_control sample

gpx1000 committed:
update copyrights.

asuessenbach commented on framework/vulkan_type_mapping.h:
```diff
@@ -23,368 +23,21 @@
-	using Type = vk::TensorViewARM;
-};
-}        // namespace detail
-
```
Is this gone due to merging?

gpx1000 replied:
This is no longer part of the PR.  it should e gone due to merging.

gpx1000 committed:
Merge branch 'main' into shader_quad_control

gpx1000 committed:
Merge branch 'main' into shader_quad_control
